Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift

```diff
@@ -44,40 +44,45 @@ class ViewController: UIViewController {
     }
     
     @IBAction func acquireToken(_ sender:UIButton) {
-        let authContext = ADAuthenticationContext(authority: "https://login.microsoftonline.com/common",
-                                                  error: nil)
         
-        authContext!.acquireToken(withResource: "https://graph.windows.net",
-                                             clientId: "b92e0ba5-f86e-4411-8e18-6b5f928d968a",
-                                             redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob"))
+        guard let authContext = ADAuthenticationContext(authority: "https://login.microsoftonline.com/common", error: nil) else {
+            
+            print("Failed to create auth context")
+            return
+        }
+        
+        authContext.acquireToken(withResource: "https://graph.windows.net",
+                                 clientId: "b92e0ba5-f86e-4411-8e18-6b5f928d968a",
+                                 redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob")!)
         {
             (result) in
             
-            if (result!.status != AD_SUCCEEDED)
-            {
-                if result!.error.domain == ADAuthenticationErrorDomain
-                    && result!.error.code == ADErrorCode.ERROR_UNEXPECTED.rawValue {
-                    
-                    self.updateStatusField("Unexpected internal error occured");
-                    
-                } else {
+            guard result.status == AD_SUCCEEDED else {
+                
+                if result.error!.domain == ADAuthenticationErrorDomain
+                    && result.error!.code == ADErrorCode.ERROR_UNEXPECTED.rawValue {
                     
-                    self.updateStatusField(result!.error.description)
+                    self.updateStatusField("Unexpected internal error occured")
```

Comment:
check this method calls ui update on the main thread

Reply:
Addressed